### PR TITLE
Fix double navigate modal screen

### PR DESCRIPTION
**Pull request by evshi**
## Issue ticket number and link

Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have added tests
- [ ] I have added a changeset `pnpm changeset add`
- [ ] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/runtime/src/runtime/modal/index.tsx [hunk 1]
```diff
@@ -142,23 +163,24 @@ export const ModalWrapper: React.FC<PropsWithChildren> = ({ children }) => {
...
             return oldModalState;
           }
-          return [
+          const nextState = [
             ...oldModalState.slice(0, -1),
             {
               ...oldModalState[oldModalState.length - 1],
               visible: false,
             },
           ];
```
Comment (evshi) [inline packages/runtime/src/runtime/modal/index.tsx:172]: @anserwaseem what was the purpose of this?

Comment (anserwaseem) [reply packages/runtime/src/runtime/modal/index.tsx:172]: it's there such that when a new modal is opened via `navigateModalScreen`, the previous one is hidden

**Comment by anserwaseem** [comment]
what is the exact issue here? is modal created twice in DOM tree?

**Comment by evshi** [comment]
Yes, this triggers two renders of the modal wrapper and causes the modal to be added twice.

**Review by anserwaseem — APPROVED** [review]

**Pull request merged**